src/__tests__/upload-packages.test.ts

```diff
@@ -0,0 +1,96 @@
+import { uploadPackages } from '../upload-packages';
+import * as actionsCore from '@actions/core';
+import * as uploadPackageArtifact from '../features/upload-package-artifact';
+import { mockPartial } from '../utils/mocks';
+import { when } from 'jest-when';
+
+jest.mock('@actions/core');
+jest.mock('../features/upload-package-artifact');
+
+describe('upload packages', () => {
+  const getInputSpy = jest.spyOn(actionsCore, 'getInput');
+  const setOutputSpy = jest.spyOn(actionsCore, 'setOutput');
+  const setFailedSpy = jest.spyOn(actionsCore, 'setFailed');
+
+  const uploadPackageArtifactSpy = jest.spyOn(uploadPackageArtifact, 'uploadPackageArtifact');
+
+  it('should parse inputs and set outputs', async () => {
```

Comment:
I would prefer to split this test. It is too abstract, to mane `expect` statements. Tests are not isolated and affects with each other.

Reply:
One test per code branch

Reply:
what is `code branch`?